snykTask/src/index.ts

```diff
@@ -179,9 +194,34 @@ async function runSnykTest(
   if (snykTestExitCode >= CLI_EXIT_CODE_INVALID_USE) {
     code = snykTestExitCode;
     errorMsg =
-      'failing task because `snyk test` was improperly used or had other errors';
+      'failing task because `snyk` was improperly used or had other errors';
   }
   const snykOutput: SnykOutput = { code: code, message: errorMsg };
+
+  // handle snyk code no-issues-found non-existent json by rerunning --json stdout to piped file
+  if (
+    taskArgs.testType == TestType.CODE &&
+    !fs.existsSync(jsonReportOutputPath) &&
+    snykTestExitCode === CLI_EXIT_CODE_SUCCESS
+  ) {
+    const echoToolRunner = tl.tool('echo');
+    const snykCodeTestToolRunner = tl
+      .tool(snykPath)
+      .arg('code')
+      .arg('test')
+      .arg('--json')
+      .argIf(
+        taskArgs.codeSeverityThreshold,
+        `--severity-threshold=${taskArgs.codeSeverityThreshold}`,
+      )
+      .argIf(taskArgs.ignoreUnknownCA, `--insecure`)
+      .argIf(taskArgs.organization, `--org=${taskArgs.organization}`)
+      .argIf(taskArgs.projectName, `--project-name=${projectNameArg}`)
+      .line(taskArgs.additionalArguments)
+      .pipeExecOutputToTool(echoToolRunner, jsonReportOutputPath);
+    await snykCodeTestToolRunner.exec(options);
+  }
```

Comment:
I've searched for the usage of `additionalArguments` and it seems that it might not be in use anymore. Could you please help me to understand if this is correct? Maybe this is an opportunity to remove deprecated code and if it exists, maybe we could add to the README a link to the docs that shows how to use it. WDYT?